pkg/git/libgit2/managed/http.go

```diff
@@ -279,7 +279,7 @@ func createClientRequest(targetURL string, action git2go.SmartServiceAction,
 
 	// Apply authentication and TLS settings to the HTTP transport.
 	if authOpts != nil {
-		if len(authOpts.Username) > 0 {
+		if authOpts.Username != "" && authOpts.Password != "" {
```

Comment:
```suggestion
		if authOpts.Username != "" || authOpts.Password != "" {
```

Reply:
We only want to set basic auth when both username and password are non-blank. If the username is not blank but the password is, the OR operator will let this if condition match and we will set basic auth, which will lead to a 401.